src/core/core.ts

```diff
@@ -387,6 +397,40 @@ class Zoomist {
     this.emit('wheel', this, this.transform.scale, e)
   }
 
+  #useDblClick(e: MouseEvent) {
+    e.preventDefault();
+    setStyle(this.image, {transition: 'transform ease 0.3s'});
+    const {options: {dbClickZoomRatio}} = this
+    if (this.isOnMinScale()) {
+      this.zoom(dbClickZoomRatio, getPointer(e))
+    } else {
+      this.zoom(-1);
+    }
+    this.emit('dblClick', this, this.transform.scale, e)
+    setTimeout(() => setStyle(this.image, {transition: ''}), 300)
+  }
+
+  #useDblTouch(e: TouchEvent) {
+    e.preventDefault();
+    if (e.touches.length !== 1) {
+      return;
+    }
+    if (Date.now() - this.data.dblTouchData.lastTouchTime < 300) {
+      setStyle(this.image, {transition: 'transform ease 0.3s'});
+      const {options: {dbClickZoomRatio}} = this
+      if (this.isOnMinScale()) {
+        this.zoom(dbClickZoomRatio, getPointer(e))
+      } else {
+        this.zoom(-1);
+      }
+      this.data.dblTouchData.lastTouchTime = 0;
+      this.emit('dblTouch', this, this.transform.scale, e);
```

Comment:
I am not so sure to introduce separate event since there is only one setting enabling the feature for mobile and desktop

Reply:
Yeah, I think we should keep the API consistent, either split mobile from desktop or merge them. I'd say merge to keep it simple, until we have a good reason to complicate.